packages/vue/src/sdk.ts

```diff
@@ -0,0 +1,416 @@
+  /**
+   * List of hooks to keep track of during component lifecycle.
+   * Available hooks: 'activate' | 'create' | 'destroy' | 'mount' | 'update'
+   * Based on https://vuejs.org/v2/api/#Options-Lifecycle-Hooks
+   */
+  hooks?: Operation[];
+
+  /** {@link TracingOptions} */
+  tracingOptions: TracingOptions;
+}
+
+/** Global Vue object limited to the methods/attributes we require */
+interface VueInstance {
+  config: {
+    errorHandler?(error: Error, vm?: ViewModel, info?: string): void;
+  };
+  util?: {
+    warn(...input: any): void;
+  };
+  mixin(hooks: { [key: string]: () => void }): void;
+}
+
+/** Representation of Vue component internals */
+interface ViewModel {
+  [key: string]: any;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  $root: object;
+  $options: {
+    [key: string]: any;
+    name?: string;
+    propsData?: { [key: string]: any };
+    _componentTag?: string;
+    __file?: string;
+    $_sentryPerfHook?: boolean;
+  };
+  $once(hook: string, cb: () => void): void;
+}
+
+/** Optional metadata attached to Sentry Event */
+interface Metadata {
+  [key: string]: any;
+  componentName?: string;
+  propsData?: { [key: string]: any };
+  lifecycleHook?: string;
+}
+
+// https://vuejs.org/v2/api/#Options-Lifecycle-Hooks
+type Hook =
+  | 'activated'
+  | 'beforeCreate'
+  | 'beforeDestroy'
+  | 'beforeMount'
+  | 'beforeUpdate'
+  | 'created'
+  | 'deactivated'
+  | 'destroyed'
+  | 'mounted'
+  | 'updated';
+
+type Operation = 'activate' | 'create' | 'destroy' | 'mount' | 'update';
+
+// Mappings from operation to corresponding lifecycle hook.
+const HOOKS: { [key in Operation]: Hook[] } = {
+  activate: ['activated', 'deactivated'],
+  create: ['beforeCreate', 'created'],
+  destroy: ['beforeDestroy', 'destroyed'],
+  mount: ['beforeMount', 'mounted'],
+  update: ['beforeUpdate', 'updated'],
+};
+
+const COMPONENT_NAME_REGEXP = /(?:^|[-_/])(\w)/g;
+const ROOT_COMPONENT_NAME = 'root';
+const ANONYMOUS_COMPONENT_NAME = 'anonymous component';
+
+/** Vue specific configuration for Tracing Integration  */
+interface TracingOptions {
+  /**
+   * Decides whether to track components by hooking into its lifecycle methods.
+   * Can be either set to `boolean` to enable/disable tracking for all of them.
+   * Or to an array of specific component names (case-sensitive).
+   */
+  trackComponents: boolean | string[];
+  /** How long to wait until the tracked root activity is marked as finished and sent of to Sentry */
+  timeout: number;
+  /**
+   * List of hooks to keep track of during component lifecycle.
+   * Available hooks: 'activate' | 'create' | 'destroy' | 'mount' | 'update'
+   * Based on https://vuejs.org/v2/api/#Options-Lifecycle-Hooks
+   */
+  hooks: Operation[];
+}
+
+/**
+ * Inits the Vue SDK
+ */
+export function init(
+  options: Partial<Omit<VueOptions, 'tracingOptions'> & { tracingOptions: Partial<TracingOptions> }> = {},
+): void {
+  if (options.defaultIntegrations === undefined) {
```

Comment:
`options.defaultIntegrations === true` should also do this.

Reply:
It can also be an array, we have this logic everywhere so I rather not change it just here.

Reply:
Eh, then we have a bug in 4 places :(

```js
Sentry.init({
  defaultIntegrations: true
})
```

should work, but it won't now. Can we fix it in this PR plz?

Reply:
The type of the option is defined as `defaultIntegrations?: false | Integration[];`
So either it has to be omitted, or false or array. It can't be set to `true`